examples/nucleo_l452re/lvgl/main.cpp

```diff
@@ -0,0 +1,218 @@
+
+#include <lvgl/lvgl.h>
+
+
+// Set the log level
+#undef	MODM_LOG_LEVEL
+#define	MODM_LOG_LEVEL modm::log::DEBUG
+
+using namespace Board;
+using namespace modm::literals;
+
+static modm::Abandonment
+test_assertion_handler(const modm::AssertionInfo &info)
+{
+	MODM_LOG_ERROR << "AssertionInfo.name: " << info.name << "\n";
+	MODM_LOG_ERROR << "AssertionInfo.context: " << info.context << "\n";
+	MODM_LOG_ERROR << "AssertionInfo.behavior: " << info.behavior << "\n";
+	MODM_LOG_ERROR << modm::endl;
+	return modm::Abandonment::DontCare;
+}
+MODM_ASSERTION_HANDLER(test_assertion_handler);
+
+
+namespace tft
+{
+	using DmaRx = Dma1::Channel2;
+	using DmaTx = Dma1::Channel3;
+	using Spi = SpiMaster1_Dma<DmaRx, DmaTx>;
+	//using Spi = SpiMaster1;
+	using Cs = modm::platform::GpioC8;
+	using Sck = modm::platform::GpioA5;
+	using Miso = modm::platform::GpioA6;
+	using Mosi = modm::platform::GpioA7;
+	using DataCommands = modm::platform::GpioC5;
+	using Reset = modm::platform::GpioC6;
+	using Backlight = modm::platform::GpioC9;
+}
+
+modm::Ili9341Spi<
+	tft::Spi,
+	tft::Cs,
+	tft::DataCommands,
+	tft::Reset,
+	tft::Backlight
+> tftController;
+
+namespace touch
+{
+	using Spi = SpiMaster2;
+	using Cs = modm::platform::GpioB3;
+	using Sck = modm::platform::GpioB13;
+	using Miso = modm::platform::GpioB14;
+	using Mosi = modm::platform::GpioB15;
+	//using Interrupt = modm::platform::GpioA10;
+}
+
+modm::Touch2046<touch::Spi, touch::Cs> touchController;
+
+
+static lv_disp_buf_t disp_buf;
+static constexpr size_t buf_size = LV_HOR_RES_MAX * LV_VER_RES_MAX / 8;
+static lv_color_t buf[buf_size];
+
+bool my_touchpad_read(lv_indev_drv_t* indev, lv_indev_data_t* data)
+{
+	data->state = RF_CALL_BLOCKING(touchController.isTouched()) ? LV_INDEV_STATE_PR : LV_INDEV_STATE_REL;
+	if(data->state == LV_INDEV_STATE_PR) {
+		auto xy = RF_CALL_BLOCKING(touchController.getTouchPosition());
+		data->point.x = std::get<0>(xy);
+		data->point.y = std::get<1>(xy);
+	}
+	return false;
+}
+
+void my_flush_cb(lv_disp_drv_t * disp_drv, const lv_area_t * area, lv_color_t * color_p)
+{
+	tftController.drawRaw(
+		{area->x1, area->y1},
+		(area->x2 - area->x1 +1),
+		(area->y2 - area->y1 + 1),
+		(modm::glcd::Color*)color_p);
+	lv_disp_flush_ready(disp_drv);
+}
+
+void my_log_cb(lv_log_level_t level, const char * file, uint32_t line, const char * fn_name, const char * dsc)
+{
+	if(level == LV_LOG_LEVEL_ERROR) {
+		MODM_LOG_ERROR << "LVGL(" << file << ":" << line << ") " << fn_name << ": " << dsc << modm::endl;
+	}
+	else if(level == LV_LOG_LEVEL_WARN) {
+		MODM_LOG_WARNING << "LVGL(" << file << ":" << line << ") " << fn_name << ": " << dsc << modm::endl;
+	}
+	else if(level == LV_LOG_LEVEL_INFO) {
+		MODM_LOG_INFO << "LVGL(" << file << ":" << line << ") " << fn_name << ": " << dsc << modm::endl;
+	}
+	else if(level == LV_LOG_LEVEL_TRACE) {
+		MODM_LOG_DEBUG << "LVGL(" << file << ":" << line << ") " << fn_name << ": " << dsc << modm::endl;
+	}
+}
```

Comment:
Could this be moved into the `:lvgl` module? This seems very useful to have in general.

Reply:
Yes, but I'm unsure how to implement it.
I don't want to modify the `lv_init()` code, but we could move the code into a `lv_modm_logging.cpp` file and offer a function `lv_register_modm_logging()` to the user.
Other ideas?

Reply:
You could put it into the `.init` array, so it gets called like a static constructor. This only works if you can register this before LVGL initialization.